pkgs/applications/networking/cluster/terraform/default.nix

```
@@ -62,21 +62,24 @@ let
 
           # Make providers available in Terraform 0.13 and 0.12 search paths.
           pluginDir = lib.concatMapStrings (pl: let
-            inherit (pl) repo version GOOS GOARCH;
-            inherit (pl.passthru) providerSourceAddress;
+            inherit (pl) version GOOS GOARCH;
+
+            pname = pl.pname or (throw "${pl.name} is missing a pname attribute");
+
+            slug = pl.passthru.providerSourceAddress or "nixpkgs/${pname}";
```

```suggestion
            slug = pl.passthru.provider-source-address or "nixpkgs/${pname}";
```

The key is in kebab case in the JSON (not sure that was wise on my part).

---

`"nixpkgs/${pname}"` lacks the provider name in the path.

Example with `pname = "terraform-provider-auth0"`. The provider source address becomes `nixpkgs/terraform-provider-auth0` where it should be ideally be `nixpkgs/terraform-provider/auth0`.

`"nixpkgs/${pname}/${pname}"` most definitely doesn't work:
```
Error: Invalid provider type

  on main.tf line 7, in terraform:
   7:     auth0 = {
   8:       source  = "nixpkgs/terraform-provider-auth0/terraform-provider-auth0"
   9:       version = "0.11.0"
  10:     }

Provider source "nixpkgs/terraform-provider-auth0/terraform-provider-auth0"
has a type with the prefix "terraform-provider-", which isn't valid. Although
that prefix is often used in the names of version control repositories for
Terraform providers, provider source strings should not include it.

Did you mean "nixpkgs/terraform-provider-auth0/auth0"?
```
What about we strip `terraform-provider-` for the last segment of the path?


`"nixpkgs/${pname}/${replaceStrings ["terraform-provider-"] [""] pname}"`.

---

Would this work?
```
auth0 = {
  source  = "nixpkgs/auth0"
  version = "0.11.0"
}
```

---

It would provided the provided is placed inside the `registry.terraform.io` host.

```nix
"registry.terraform.io/nixpkgs/${lib.replaceStrings ["terraform-provider-"] [""] pname}"
```

When the host section of provider source address is omitted, it will default to `registry.terraform.io`.